_posts/2020-12-07-Pivoting.md

```diff
@@ -0,0 +1,85 @@
+Cluster-API and Metal3 objects from the ephemeral k8s cluster to a target
+cluster. In Metal3, this process is performed using the
+[clusterctl](https://cluster-api.sigs.k8s.io/clusterctl/overview.html) tool
+provided by Cluster-API. clusterctl recognizes pivoting as move. During the
+pivot process clusterctl pauses any reconciliation of Cluster-API objects and
+this gets propagated to Cluster-api-provider-metal3 (CAPM3) objects as well.
+Once all the objects are paused, the objects are created on the other side on
+the target cluster and deleted from the ephemeral cluster.
+
+## Prerequisites
+
+Prior to the actual pivot process, target cluster should already have the
+provider components, ironic containers and CNI installed and running.
+Specifically the following points need to be addressed:
+
+* clusterctl is used to initialize both the ephemeral and target cluster.
+* BMH objects have correct status annotation.
+* Maintain connectivity towards provisioning network.
+* Baremetal Operator(BMO) is deployed as part of CAPM3.
+* Objects should have a proper owner reference chain.
+
+For a detailed explanation of the above mentioned prerequisites please read the
+[pivoting documentation](https://github.com/metal3-io/metal3-docs/blob/master/docs/move.md).  
+
+## Pivoting workflow in CI
+
+The Metal3 CI currently includes the pivoting as part of the deployment
+process both for Ubuntu and CentOS based jobs. This essentially means all
+the PRs that go in, are tested through the pivoting workflow. Here is the
+CI deployment workflow:
+
+* `make` the [metal3-dev-env](https://github.com/metal3-io/metal3-dev-env.git).
+It gives us the ephemeral cluster with all the  necessary controllers running
+within it.
+* `provision` target cluster. For a normal integration tests, this step deploys
+a control-plane node and a worker in the target cluster. For, `feature-test`
+and `feature-test-upgrade` the provision step deploys three control-planes and a
+worker.
+* Initialize the provider components on the target cluster. This installs all
+the controllers and associated components related to cluster-api ,
+cluster-api-provider-metal3, baremetal-operator and ironic. Since it is
+necessary to have only one set of ironic deployment/containers in the picture,
+this step also deletes the ironic deployment/containers from
+ephemeral cluster.
+* `Move` all the objects from ephemeral to target cluster.
+* Check the status of the objects to verify whether the objects are being
+reconciled correctly by the controllers in the target cluster. This step
+verifies and finalizes the pivoting process.
+* `Move` the objects back to the ephemeral cluster. This step also
+removes the ironic deployment from target cluster and reinstates the
+ironic deployment/containers in the ephemeral cluster. Since we do
+not delete the provider components in the ephemeral cluster,
+installing them again is not necessary.
+* `De-provision` the BMHs and delete the target cluster.
```

Comment:
Some here are commands "make", others concepts. It would be great if for each step an example command could be given (through makefile or by calling the ansible playbooks)